src/Deserializers/LegacyEntityIdDeserializer.php

```diff
@@ -35,7 +35,7 @@ public function deserialize( $serialization ) {
 		if ( is_string( $serialization ) ) {
 			return $this->getParsedId( $serialization );
 		}
-		elseif ( $this->isLegacyFormat( $serialization ) ) {
+		elseif ( $this->isDeserializerFor( $serialization ) ) {
```

Comment:
This is not really the correct usage of the interface. `isDeserializerFor` should always return true when it can deserialize the given serialization. However, in this case it returns false for strings but might still be able to deserialize them.

Furthermore, I think that we should use the Dispatchable interface only when we actually use the Deserializer in a DispatchingDeserializer. If no dispatching happens it's kind of pointless to implement the interface.